apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -254,56 +274,56 @@ func (r *OpenStackControlPlane) ValidateUpdateServices(old OpenStackControlPlane
 	errors = append(errors, r.ValidateServiceDependencies(basePath)...)
 
 	// Call internal validation logic for individual service operators
-	if r.Spec.Keystone.Enabled {
-		errors = append(errors, r.Spec.Keystone.Template.ValidateUpdate(old.Keystone.Template, basePath.Child("keystone").Child("template"))...)
+	if r.Spec.Keystone.Enabled && r.Spec.Keystone.Template != nil && old.Keystone.Template != nil {
```

Comment:
here and below: When the request chages the template from nil to non-nil we still need to validate the content of the non-nil template.

//later

As the defaulting webhook ensures that the template is never nil we should simply reject the Update if either the old or the new template is nil that simply should not happen
Also as we are defaulting the template even if the service is disabled we should be able to validate it when the service is disabled.

Reply:
as mentioned above setting the defaulting webhook makes the Template non nil, but this is not stored, so on an update the old template is still nil.